Sources/TidepoolKit/TAPI.swift

```diff
@@ -238,14 +314,13 @@ public class TAPI {
     /// - Parameters:
     ///   - userId: The user id for which to get the profile. If no user id is specified, then the session user id is used.
     ///   - completion: The completion function to invoke with any error.
-    public func getProfile(userId: String? = nil, completion: @escaping (Result<TProfile, TError>) -> Void) {
+    public func getProfile(userId: String? = nil) async throws -> TProfile {
```

Comment:
[general comment] it seem comments for a few functions have not be updated to align with the changes. Please review all and update.